tests/stdlib/trstgen.nim

```diff
@@ -1259,6 +1259,55 @@ Test1
     let refline = "Ref. " & ref1 & "! and " & ref2 & ";and " & ref3 & "."
     doAssert refline in output1
 
+  test "Option lists 1":
+    # check that "* b" is not consumed by previous bullet item because of
+    # incorrect indentation handling in option lists
+    let input = dedent """
+      * a
+        -m   desc
+        -n   very long
+             desc
+      * b"""
+    let output = input.toHtml
+    check(output.count("<ul") == 1)
+    check(output.count("<li>") == 2)
+    check(output.count("<table") == 1)
+    check("""<th align="left">-m</th><td align="left">desc</td>""" in output)
+    check("""<th align="left">-n</th><td align="left">very long desc</td>""" in
+          output)
+
+  test "Option lists 2":
+    # check that 2nd option list is not united with the 1st
+    let input = dedent """
+      * a
+        -m   desc
+        -n   very long
+             desc
+      -d  option"""
+    let output = input.toHtml
+    check(output.count("<ul") == 1)
+    check(output.count("<table") == 2)
+    check("""<th align="left">-m</th><td align="left">desc</td>""" in output)
```

Comment:
not a blocker but a comment/question: isn't there an intermediate representation we could use for such checks instead of going straight from input rst to output html?
(eg: something like macros.treeRepr but for rst)

Reply:
@timotheecour interesting idea.

We could use `renderRstToStr` from rstast.nim, it basically returns AST representation in the string form.

However I think it's reasonable to compare with HTML (or Latex):
- we test both parsing and generation
- we don't depend on our own intermediate generation

There would be a downside to fix tests too many times if we were changing `rstgen.nim` constantly. But rstgen.nim usually stays fixed.

Reply:
the AST tests are easier to understand in code, whereas the html output is easier to understand when rendered in a browser, so maybe we can have both kinds of tests:
* AST based tests for most unittests
* a single (or few) integration test that test the final html rendering; this wouldn't be written by hand but rendered in a groundtruth file that's checked in, and the the tester would compare against that, basically extending the tests from nimdoc/tester.nim or nimdoc/rsttester.nim

benefit:
* easy to read/write/edit unittests
* sanity check that final html stays same as groundtruth, but when we need to change thing, we don't edit the groundtruth by hand, we re-generate it and check that the new groundtruth is as we expect (both by looking at html + rendered in webpage)

Reply:
=> followup here: https://github.com/timotheecour/Nim/issues/676